bundles/org.openhab.binding.plex/src/main/java/org/openhab/binding/plex/internal/handler/PlexApiConnector.java

```diff
@@ -198,28 +204,60 @@ public boolean getApi() {
                 }
             }
             return false;
-        } catch (IOException e) {
+        } catch (IOException | InterruptedException | TimeoutException | ExecutionException e) {
             logger.debug("An exception occurred while fetching API :'{}'", e.getMessage(), e);
         }
         return false;
     }
 
     /**
-     * Make an HTTP request and return the class object that was used when calling.
+     * Make an HTTP request and return the response as a string.
      *
-     * @param <T> Class being used(dto)
      * @param method GET/POST
      * @param url What URL to call
      * @param headers Additional headers that will be used
-     * @param type class type for the XML parsing
-     * @return Returns a class object from the data returned by the call
+     * @param verify Flag to indicate if ssl certificate checking should be done
+     * @return Returns a string of the http response
      * @throws IOException
+     * @throws ExecutionException
+     * @throws TimeoutException
+     * @throws InterruptedException
      */
-    private <T> T doHttpRequest(String method, String url, Properties headers, Class<T> type) throws IOException {
-        String response = HttpUtil.executeUrl(method, url, headers, null, null, REQUEST_TIMEOUT_MS);
+    private String doHttpRequest(String method, String url, Properties headers, boolean verify)
+            throws IOException, InterruptedException, TimeoutException, ExecutionException {
+        final String response;
+        if (verify) {
+            // Requests sent to the PLEX servers should use certificate checking
+            response = HttpUtil.executeUrl(method, url, headers, null, null, REQUEST_TIMEOUT_MS);
```

Comment:
Don't forget, when connecting to the local plex servers the certificates are self-signed so you need to make sure you don't over-secure against those.  It's also possible that the plex server could be reached through a hostname (e.g. if it's behind an NGINX server or something similar).  The only things that should check cert IMO is if you are going to *.plex.tv and even then, it can't be the DNS redirected plex server.

Reply:
> The only things that should check cert IMO is if you are going to *.plex.tv and even then, it can't be the DNS redirected plex server.

That is what was implemented. Connections going to plex.tv still use HttpUtil.executeUrl() which will check certificates. Connections going to the local server now use the custom httpClient with certificate checking turned off [setTrustAll(true)].

Reply:
All right, let's keep it as is without further refactoring also considering https://github.com/openhab/openhab-addons/pull/15306#discussion_r1275254393.